src/NuGet.Core/NuGet.Commands/Internal/PackageSearchResultTable.cs

```diff
@@ -0,0 +1,121 @@
+        private int[] _columnsToHighlight;
+
+        public PackageSearchResultTable(int[] columnsToHighlight, params string[] headers)
+        {
+            _columnsToHighlight = columnsToHighlight;
+            _columnWidths = new int[headers.Length];
+
+            for (int i = 0; i < headers.Length; i++)
+            {
+                _columnWidths[i] = headers[i].Length;
+            }
+
+            _rows.Add(headers);
+        }
+
+        public void AddRow(params string[] row)
+        {
+            if (row.Length != _columnWidths.Length)
+            {
+                throw new InvalidOperationException("Row column count does not match header column count.");
+            }
+
+            for (int i = 0; i < row.Length; i++)
+            {
+                _columnWidths[i] = Math.Max(_columnWidths[i], row[i]?.Length ?? 0);
+            }
+
+            _rows.Add(row);
+        }
+
+        public void PrintResult(string searchTerm = null)
+        {
+            ConsoleColor highlighterColor = GetHighlighterColor();
+
+            // If only headers are present (i.e., no package rows)
+            if (_rows.Count <= 1)
+            {
+                Console.WriteLine("No results found.");
+                return;
+            }
+
+            foreach (var row in _rows)
+            {
+                for (int i = 0; i < row.Length; i++)
+                {
+                    var paddedValue = (row[i] ?? string.Empty).PadRight(_columnWidths[i]);
+
+                    if (!string.IsNullOrEmpty(searchTerm) && paddedValue.IndexOf(searchTerm, StringComparison.OrdinalIgnoreCase) >= 0)
+                    {
+                        Console.Write("| ");
+                        if (_columnsToHighlight.Contains(i)) PrintWithHighlight(paddedValue, searchTerm, highlighterColor);
+                        else Console.Write(paddedValue);
+                        Console.Write(" ");
+                    }
+                    else
+                    {
+                        Console.Write("| " + paddedValue + " ");
+                    }
+                }
+
+                Console.WriteLine("|");
+
+                if (row == _rows.First())
+                {
+                    // Add the separator after the header.
+                    foreach (var width in _columnWidths)
+                    {
+                        Console.Write("|" + new string('-', width + 2));
+                    }
+
+                    Console.WriteLine("|");
+                }
+            }
+        }
+
+        private static void PrintWithHighlight(string value, string searchTerm, ConsoleColor highlighterColor)
+        {
+            int index = value.IndexOf(searchTerm, StringComparison.OrdinalIgnoreCase);
+            ConsoleColor originalColor = Console.ForegroundColor;
+
+            while (index != -1)
+            {
+                Console.Write(value.Substring(0, index));
+                Console.ForegroundColor = highlighterColor;
```

Comment:
Have you tested this when redirecting output to a file?  `dotnet package search some keywords > out.txt`?  This is effectively how CI build systems work, so if the output contains control characters, then customers who run the command in a build script could result in difficult to read output. Also, anyone who writes a shell script and saves the program output to a variable, or just pipes directly to commands like `grep` or `sed`.

Reply:
I tested `dotnet package search some keywords > out.txt` and it looks like there is no control characters generated.